_posts/2019-08-01-projeto-pendulo.md

```diff
@@ -0,0 +1,409 @@
+
+O subespaço a escolher depende do sistema e da análise desejada. No nosso caso, vamos plotar apenas os pontos que correspondem a instantes que satisfazem a relação $$\omega_F t = 2n\pi$$ onde $$n$$ é um inteiro. Isto é, estamos plotando apenas os pontos onde a trajetória está em fase com a força externa aplicada. Devido às aproximações numéricas, devemos na verdade plotar um ponto se ele satisfizer $$|t - 2n\pi/\omega_F| < \Delta t/2$$ para algum valor de $$\Delta t$$ apropriado. No nosso caso, usaremos para $$\Delta t$$ o valor do intervalo entre amostras no sistema simulado, visto que não há precisão temporal maior do que essa em nossos resultados.
+
+Vamos fazer uma função que dado um array de valores de tempo, o tempo de ciclo e a precisão de tempo $$\Delta t$$ seleciona todos os índices (retornando um vetor de booleanos) que estão sincronizados com o ciclo de acordo com a expressão acima:
+
+~~~ python
+def select_synchronized(t, Tf, Δt):
+    return np.abs(np.mod(t+Tf/2, Tf) - Tf/2) < Δt/2
+
+q = 0.5
+ncycles = 200
+tmax, nintervals, Δt = timing(ncycles, Tf)
+
+f = 0.1
+
+t, xy = simulate_forced(θ0, tmax, nintervals, g, ℓ, q, f, ωf)
+traj = normalize_angle(xy[:, 0])
+
+selected = select_synchronized(t, Tf, Δt)
+
+plt.plot(traj[selected], xy[selected, 1], 'o', markersize=3, alpha=0.3) # Plota apenas esses pontos
+plt.show()
+~~~
+
+Note como, com exceção de alguns pontos devidos ao transiente inicial, os pontos agora se acumulam em um mesmo local. A razão é simples: depois do transiente inicial, o sistema assume uma trajetória periódica de acordo com a frequência da força externa, e portanto, se olhamos a intervalos determinados pelo período dessa oscilação, encontraremos o pêndulo sempre no mesmo ponto.
+
+Vejamos agora o que acontece no caso caótico.
+
+~~~ python
+f = 1.2
+
+t, xy = simulate_forced(θ0, tmax, nintervals, g, ℓ, q, f, ωf)
+traj = normalize_angle(xy[:, 0])
+
+selected = select_synchronized(t, Tf, Δt)
+
+plt.plot(traj[selected], xy[selected, 1], '.', markersize=3, alpha=0.3)
+plt.show()
+O que acontece se deixamos o sistema executar por mais tempo?
+
+ncycles = 2000
+tmax, nintervals, Δt = timing(ncycles, Tf)
+
+t, xy = simulate_forced(θ0, tmax, nintervals, g, ℓ, q, f, ωf)
+
+traj = normalize_angle(xy[:, 0])
+selected = select_synchronized(t, Tf, Δt)
+
+plt.plot(traj[selected], xy[selected, 1], '.', markersize=3, alpha=0.3)
+plt.show()
+~~~
+
+Note como o formato ficou o mesmo. Quer dizer, apesar de toda a característica caótica do sistema, ele é atraído a uma região específica do espaço de estado, denominado um **atrator**. A estrutura dos atratores no regime caótico é bastante complexa, tendo dimensionalidades fracionárias (o número de dimensões do espaço do atrator não é um número inteiro), e portanto são denominados **fractais**; os atratores fractais são denominados de **atratores estranhos**.
```

Comment:
pode trocar os negritos por itálicos? Negrito não deve ser usado no meio do texto